aioxmpp/presence/service.py

```diff
@@ -236,6 +236,183 @@ def handle_presence(self, st):
                 self.on_changed(st.from_, st)
 
 
+class DirectedPresenceHandle:
+    """
+    Represent a directed presence relationship with a peer.
+
+    Directed Presence is specified in :rfc:`6121` section 4.6. Since the users
+    server is not responsible for distributing presence updates to peers to
+    which the client has sent directed presence, special handling is needed.
+    (The only presence automatically sent by a client’s server to a peer which
+    has received directed presence is the
+    :attr:`~aioxmpp.PresenceType.UNAVAILABLE` presence which is created when
+    the client disconnects.)
+
+        .. note::
+
+            Directed presence relationships get
+            :meth:`unsubscribed <unsubscribe>` immediately when the stream is
+            destroyed. This is because the peer has received
+            :attr:`~aioxmpp.PresenceType.UNAVAILABLE` presence from the client’s
+            server.
+
+    .. autoattribute:: address
+
+    .. autoattribute:: muted
+
+    .. autoattribute:: presence_filter
+
+    .. automethod:: set_muted
+
+    .. automethod:: unsubscribe
+
+    .. automethod:: send_presence
+    """
+
+    def __init__(self, service: "PresenceServer", peer: aioxmpp.JID,
+                 muted: bool = False):
+        super().__init__()
+        self._address = peer
+        self._service = service
+        self._muted = muted
+        self._presence_filter = None
+        self._unsubscribed = False
+
+    def _require_subscribed(self):
+        if self._unsubscribed:
+            raise RuntimeError("directed presence relationship is unsubscribed")
+
+    @property
+    def address(self) -> aioxmpp.JID:
+        """
+        The address of the peer. This attribute is read-only.
+
+        To change the address of a peer,
+        :meth:`aioxmpp.PresenceServer.rebind_directed_presence` can be used.
+        """
+        return self._address
+
+    @property
+    def muted(self) -> bool:
+        """
+        Flag to indicate whether the directed presence relationship is *muted*.
+
+        If the relationship is **not** muted, presence updates made through the
+        :class:`PresenceServer` will be unicast to the peer entity of the
+        relationship.
+
+        For a muted relationships, presence updates will *not* be automatically
+        sent to the peer.
+
+        The *muted* behaviour is useful if presence updates need to be managed
+        by a service for some reason.
+
+        This attribute is read-only. It must be modified through
+        :meth:`set_muted`.
+        """
+        return self._muted
+
+    @property
+    def presence_filter(self):
```

Comment:
Kind of agreed on the nomenclature. Suggestions?

It is not a ``Filter`` object because the handle is under full control of a single object usually (in contrast to the filters on the stanza stream). If the single object needs to delegate control to multiple other things, it can do that by assigning a filter to that attribute.